tough/src/http.rs

```diff
@@ -125,93 +132,222 @@ pub struct HttpTransport {
 }
 
 /// Implement the `tough` `Transport` trait for `HttpRetryTransport`
+#[async_trait]
 impl Transport for HttpTransport {
-    /// Send a GET request to the URL. Request will be retried per the `ClientSettings`. The
-    /// returned `RetryRead` will also retry as necessary per the `ClientSettings`.
-    fn fetch(&self, url: Url) -> Result<Box<dyn Read + Send>, TransportError> {
-        let mut r = RetryState::new(self.settings.initial_backoff);
-        Ok(Box::new(
-            fetch_with_retries(&mut r, &self.settings, &url)
-                .map_err(|e| TransportError::from((url, e)))?,
-        ))
+    async fn fetch(&self, url: Url) -> Result<TransportStream, TransportError> {
+        let r = RetryState::new(self.settings.initial_backoff);
+        Ok(fetch_with_retries(r, &self.settings, &url).boxed())
+    }
+}
+
+enum RequestState {
+    /// A response is streaming.
+    Streaming(BoxStream<'static, reqwest::Result<bytes::Bytes>>),
+    /// A request is pending.
+    Executing(BoxFuture<'static, reqwest::Result<reqwest::Response>>),
```

Comment:
Better match to the description?

```suggestion
    Pending(BoxFuture<'static, reqwest::Result<reqwest::Response>>),
```